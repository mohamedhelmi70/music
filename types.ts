export type AppParamList = {
    Tab: undefined;
    Home: undefined;
    Track: {
        trackId: number;
    }
    Artists: undefined;
    Artist: {
        artistId: number;
    }
    Search: undefined;
    Albums: undefined;
    Album:  {
        albumId: number;
    }
};

export interface Album {
    album: string,
    id_album: number,
    cover: string,
    api_album: string,
    api_tracks: string
}

export interface Track {
    id_track: number;
    track: string;
    bpm: number;
    api_track: string;
    api_lyrics: string;
    artist: string;
}

export interface Artist {
    id_artist: number,
    artist: string,
    cover: string,
}