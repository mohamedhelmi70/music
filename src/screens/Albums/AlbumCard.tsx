import React from 'react';
import {StyleSheet, View as DefaultView} from 'react-native';
import {Album} from '../../../types';
import {wp} from '../../constants';
import FastImage from '../../components/Theme/FastImage';
import {Text, View} from '../../components/Themed';

interface AlbumCardProps {
    album: Album;
}

function AlbumCard({album}: AlbumCardProps) {
    return (
        <DefaultView style={styles.container}>
            <View>{Boolean(album.cover) && <FastImage url={album.cover} style={styles.image} />}</View>
            <View style={styles.cardRight}>
                <Text style={styles.cardTitle} weight="700" numberOfLines={1} lightColor="#334443">
                    {album.album}
                </Text>
            </View>
        </DefaultView>
    );
}

const styles = StyleSheet.create({
    container: {
        flex: 1,
        width: wp('45%'),
        borderRadius: wp('2%'),
        alignItems: 'center',
        marginBottom: wp('5%'),
    },
    image: {
        width: wp('40%'),
        height: wp('40%'),
        borderRadius: wp('3%'),
        marginBottom: wp('2%'),
    },
    cardRight: {
        marginLeft: wp('4%'),
    },
    cardTitle: {
        fontSize: wp('3.2%'),
        maxWidth: wp('40%'),
        width: wp('40%'),
    },
});

export default AlbumCard;
