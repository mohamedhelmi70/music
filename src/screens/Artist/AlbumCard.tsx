import React from 'react';
import {StyleSheet, Pressable} from 'react-native';
import {Album} from '../../../types';
import {colors, wp} from '../../constants';
import FastImage from '../../components/FastImage';
import {Text, View} from '../../components/Themed';

interface AlbumCardProps {
    album: Album;
    selectedAlbum: number;
    selectAlbum: (id: number) => void;
}

function AlbumCard({album, selectedAlbum, selectAlbum}: AlbumCardProps) {
    return (
        <Pressable style={styles.container} onPress={() => selectAlbum(album.id_album)}>
            <View>{Boolean(album.cover) && <FastImage url={album.cover} style={styles.image} />}</View>
            <View>
                <Text
                    style={styles.cardTitle}
                    weight="700"
                    numberOfLines={1}
                    lightColor={album.id_album === selectedAlbum ? colors.light.primary : '#334443'}>
                    {album.album}
                </Text>
            </View>
        </Pressable>
    );
}

const styles = StyleSheet.create({
    container: {
        flex: 1,
        width: wp('25%'),
        borderRadius: wp('2%'),
        alignItems: 'center',
        marginBottom: wp('5%'),
    },
    image: {
        width: wp('25%'),
        height: wp('25%'),
        borderRadius: wp('3%'),
        marginBottom: wp('2%'),
    },
    cardTitle: {
        fontSize: wp('3.2%'),
        maxWidth: wp('20%'),
        width: wp('19%'),
    },
});

export default AlbumCard;
