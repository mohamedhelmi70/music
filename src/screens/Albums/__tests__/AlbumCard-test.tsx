import React from 'react';
import renderer from 'react-test-renderer';
import {Album} from '../../../../types';
import AlbumCard from '../AlbumCard';

// Describing a test suite
describe('<AlbumCard />', () => {
    const album: Album = {
        album: 'album name',
        id_album: 1233,
        cover: 'https://image.png',
        api_album: 'hhtps://music.com/album',
        api_tracks: 'hhtps://music.com/tracks',
    };
    test('Renders AlbumCard correctly', () => {
        const tree = renderer.create(<AlbumCard album={album} />).toJSON();
        expect(tree).toMatchSnapshot();
    });
});
