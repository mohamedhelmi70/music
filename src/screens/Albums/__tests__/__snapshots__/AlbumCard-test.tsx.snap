// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<AlbumCard /> Renders AlbumCard correctly 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "borderRadius": 15,
      "flex": 1,
      "marginBottom": 37.5,
      "width": 337.5,
    }
  }
>
  <View
    style={
      Array [
        Object {
          "backgroundColor": "#F7F7F9",
        },
        undefined,
      ]
    }
  >
    <View
      style={
        Array [
          Object {
            "overflow": "hidden",
          },
          Object {
            "borderRadius": 22.5,
            "height": 300,
            "marginBottom": 15,
            "width": 300,
          },
        ]
      }
    >
      <FastImageView
        resizeMode="cover"
        source={
          Object {
            "cache": "immutable",
            "headers": Object {
              "Authorization": "4f298a53e552283bee957836a529baec",
            },
            "priority": "normal",
            "uri": "https://image.png",
          }
        }
        style={
          Object {
            "bottom": 0,
            "left": 0,
            "position": "absolute",
            "right": 0,
            "top": 0,
          }
        }
      />
    </View>
  </View>
  <View
    style={
      Array [
        Object {
          "backgroundColor": "#F7F7F9",
        },
        Object {
          "marginLeft": 30,
        },
      ]
    }
  >
    <Text
      numberOfLines={1}
      style={
        Array [
          Object {
            "color": "#334443",
            "fontFamily": "Roboto",
            "fontWeight": "700",
            "textAlign": "left",
          },
          Object {
            "fontSize": 24,
            "maxWidth": 300,
            "width": 300,
          },
        ]
      }
    >
      album name
    </Text>
  </View>
</View>
`;
